tutorials/ab3TracingCode.md

```diff
@@ -281,31 +281,224 @@ Here are some quick questions you can try to answer based on your execution path
+
+    ```java {.line-numbers highlight-lines="3"}
+    private CommandResult executeCommand(String commandText) throws CommandException, ParseException {
+        ...
+        } catch (CommandException | ParseException e) {
+            logger.info("An error occurred while executing command: " + commandText);
+            resultDisplay.setFeedbackToUser(e.getMessage());
+            throw e;
+        }
+    }
+    ```
+</panel>
+
+<panel header="2. `edit 0 n/Alice Yu`" no-close>
+
+* Exception Thrown: ParseException
+* Reason: Invalid command format as index 0 is not a non-zero unsigned integer.
+* Where the exception is thrown: `EditCommandParser#parse()`
+
+    ```java {.line-numbers highlight-lines="6"}
+    public EditCommand parse(String args) throws ParseException {
+       ...
+       try {
+           index = ParserUtil.parseIndex(argMultimap.getPreamble());
+       } catch (ParseException pe) {
+           throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+       }
+       ...
+    }
+    ```
+
+* Where the exception is handled: `MainWindow#executeCommand()`
+  </panel>
 
-    4.  `edit 1`
+<panel header="3. `edit 1 n/Alex Yeoh`" no-close>
 
-    5.  `edit 1 n/アリス ユー`
+* No exception thrown.
+* Reason: "1" is a valid index in the person list. The command is correctly formatted and will edit the name of the person at index 1 to "Alex Yeoh".
+  </panel>
 
-    6.  `edit 1 t/one t/two t/three t/one`
+<panel header="4. `edit 1`" no-close>
+
+* Exception Thrown: ParseException
+* Reason: At least one field to edit must be provided.
+* Where the exception is thrown: `EditCommandParser#parse()`
+
+    ```java {.line-numbers highlight-lines="4"}
+    public EditCommand parse(String args) throws ParseException {
+        ...
+        if (!editPersonDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
+        }
+        ...
+    }
+    ```
+
+* Where the exception is handled: `MainWindow#executeCommand()`
+  </panel>
+
+<panel header="5. `edit 1 n/アリス ユー`" no-close>
+
+* Exception Thrown: ParseException
+* Reason: Names should only contain alphanumeric characters and spaces, and it should not be blank.
+* Where the exception is thrown: `ParserUtil#parseName()`
+
+    ```java {.line-numbers highlight-lines="4"}
+    public static Name parseName(String name) throws ParseException {
+        ...
+        if (!Name.isValidName(trimmedName)) {
+            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
+        }
+        ...
+    }
+    ```
+
+* Where the exception is handled: `MainWindow#executeCommand()`
+  </panel>
+
+<panel header="6. `edit 1 t/one t/two t/three t/one`" no-close>
+
+* No exception thrown.
+* Reason: The command is correctly formatted and will edit the tags of the person at index 1 to "one", "two" and "three".
+
+    <panel header="Why didn’t the second tag with value “one” get added?">
```

Comment:
I think instead of a panel inside a panel, it would be better if the additional information was presented in an info box instead.

And the question could be made clearer: Why aren't there two tags with value "one" after this command is executed?

Lastly, would this be more accurate?
... ParserUtil#parseTags() method when tags are added to a HashSet...

What do you think?

Reply:
Thanks for the suggestion! Agreed that using an info box would be clearer for the reader. Made the changes.

Reply:
Thanks for the suggestion! Agreed that an info box would be clearer for the reader. Made the changes